wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java

```diff
@@ -176,6 +180,19 @@ public OAuthToken getToken(String oauthProvider)
       }
       if (token != null) {
         return token;
+      } else {
```

Comment:
I think we have to remove lines 174-180.
Basically don't fetch token from in memory storage at all.

Reply:
When we fetch an oauth token from SCM provider on oauth callback request, we store it in the system memory storage, but not as a secret. Right after that, when the workspace start page is reloaded by the callback request, we store the token as a secret.
So removing this block as it is will brake the oauth flow. We can rework the flow to store tokens on callback request but that requires some refactoring. I think we can do that in another iteration with a separate PR.

Reply:
This can be fixed as a part of https://github.com/eclipse/che/issues/22802